src/main/java/planner/modules/inherited/TaskWithMultipleWeeklyPeriod.java

```diff
@@ -0,0 +1,76 @@
+package planner.modules.inherited;
+
+import planner.exceptions.original.ModInvalidIndexException;
+import planner.util.periods.TimeInterval;
+import planner.util.periods.TimePeriodWeekly;
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+import java.util.List;
+
+public class TaskWithMultipleWeeklyPeriod extends TaskWithMultiplePeriods<TimePeriodWeekly> {
```

Comment:
These classes look very similar. Is there really a need to have so many types of classes?